sysl2/sysl/cmd_runner_test.go

```diff
@@ -33,3 +59,186 @@ func TestEnsureFlagsNonEmpty(t *testing.T) {
+			rootMarkerExists: true,
+		},
+		{
+			name: "Root flag is defined and root marker does not exist",
+			root: "DefinedRootAndSyslRootUndefinedTest/path/",
+			module: syslutil.MustRelative(t, "DefinedRootAndSyslRootUndefinedTest/path/",
+				definedRootNoMarker.files[0]),
+			structure:        definedRootNoMarker,
+			expectedRoot:     "DefinedRootAndSyslRootUndefinedTest/path/",
+			expectedLog:      noRootMarkerLog,
+			rootMarkerExists: false,
+		},
+		{
+			name:             "Defined relative root",
+			root:             currentWorkingDirectory,
+			module:           filepath.Clean(definedRootNoMarker.files[0]),
+			structure:        definedRootNoMarker,
+			expectedRoot:     currentWorkingDirectory,
+			expectedLog:      noRootMarkerLog,
+			rootMarkerExists: false,
+		},
+		{
+			root:             systemRoot,
+			name:             "Defined absolute path root",
+			module:           syslutil.MustAbsolute(t, definedRootNoMarker.files[0]),
+			structure:        definedRootNoMarker,
+			expectedRoot:     systemRoot,
+			expectedLog:      noRootMarkerLog,
+			rootMarkerExists: false,
+		},
+		{
+			name:             "Defined relative root with absolute module path rooted at root",
+			root:             currentWorkingDirectory,
+			module:           filepath.Join(systemRoot, filepath.Clean(definedRootNoMarker.files[0])),
+			structure:        definedRootNoMarker,
+			expectedRoot:     currentWorkingDirectory,
+			expectedLog:      noRootMarkerLog,
+			rootMarkerExists: false,
+		},
+		{
+			name:             "Defined root flag and root",
+			root:             currentWorkingDirectory,
+			module:           syslutil.MustRelative(t, currentWorkingDirectory, definedRootFlagAndMarkerFound.files[0]),
+			structure:        definedRootFlagAndMarkerFound,
+			expectedRoot:     currentWorkingDirectory,
+			expectedLog:      rootMarkerExistsLog,
+			rootMarkerPath:   syslutil.MustAbsolute(t, "./DefinedRootAndSyslRootDefinedTest/path/"),
+			rootMarkerExists: false,
+		},
+		{
+			name:             "Defined root flag and root marker with absolute path module rooted at root",
+			root:             "./DefinedRootAndSyslRootDefinedTest/",
+			module:           "/path/to/module/test.sysl",
+			structure:        definedRootFlagAndMarkerFound,
+			expectedRoot:     "./DefinedRootAndSyslRootDefinedTest/",
+			expectedLog:      rootMarkerExistsLog,
+			rootMarkerPath:   syslutil.MustAbsolute(t, "./DefinedRootAndSyslRootDefinedTest/path/"),
+			rootMarkerExists: false,
+		},
+		{
+			name:             "Root is not defined",
+			root:             "",
+			module:           undefinedRoot.files[0],
+			structure:        undefinedRoot,
+			expectedRoot:     filepath.Dir(undefinedRoot.files[0]),
+			expectedLog:      noRootLog,
+			rootMarkerExists: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(ts folderTestStructure) func(t *testing.T) {
+			return func(tt *testing.T) {
+				tt.Parallel()
```

Comment:
this could just be
```
t.Run(test.name, func(t *testing.T) {
    t.Parallel()
    ....
})  
```

Reply:
`scopelint` would give out warnings if I do that

Reply:
should be fine, thats being done in other tests in sysl

Reply:
got the warnings, ignore it?